spec/plugin.spec.js

```diff
@@ -85,7 +88,8 @@ describe('ManifestPlugin', function() {
       }, {}, function(manifest) {
         expect(manifest).toBeDefined();
         expect(manifest).toEqual({
-          'main.js': 'main.js'
+          'main.js': 'main.js',
+          'third.party.js': 'third.party.js'
```

Comment:
The purpose of this PR is to allow any third party plugin to add assets to manifest using `compilation.assets` from webpack. That is what [copy-webpack-plugin does here](https://github.com/kevlened/copy-webpack-plugin/blob/v4.0.1/src/writeFile.js#L84)

This mean that if we execute such plugins before manifest plugin executes, then manifest plugin will be able to pick those assets and add them to final output. In this case hashing i think must rely on those external plugins since files are not processed by webpack at all.

to answer the real question: `third.party.js` simulates such behaviour, the file is added to compilation* by some plugin (`copy-plugin-mock.js`) not by webpack itself when requiring/importing assets as normal.

I don't know whether i missed some edge case, so please if you think something else must be taken into account, a bit more explanation will be welcome. I have based this on finding through debugging, not through deep knowledge of webpack :)